### ISSUE TITLE
Ability to move card to another board [$20]

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/deck#249]
### Steps to reproduce
1. Create a task in a board
2. Write a lot of description into it, add tags and details etc
3. Notice it’s in the wrong board

### Expected behaviour
You can move it to the correct board.

### Actual behaviour
You can not and would need to copypaste all the details over.

Not super crucial, but might be nice to have. :)

**Comment by protocol6** [comment]
I would second this request!!!
I'm looking for a Kanban workflow that is more trello/jira-like in that tasks migrate from one stack to another ("to do" stack --> "in progress" stack --> "done" stack" )
Currently Deck doesn't seem to support that. (Together with that I'd want to use custom labels)

**Comment by juliushaertl** [comment]
> I'm looking for a Kanban workflow that is more trello/jira-like in that tasks migrate from one stack to another ("to do" stack --> "in progress" stack --> "done" stack" )
Moving inside a board is already supported.

> Together with that I'd want to use custom labels
You can also specify your own labels in the board settings. The default ones can also be changed per board.

**Comment by protocol6** [comment]
Thanks for this info! 
Also, should it be possible to move cards (horizontally) between stacks on a board?
It doesn't work for me, but maybe I'm missing something. would not be the first time. :-)

**Comment by juliushaertl** [comment]
> Also, should it be possible to move cards (horizontally) between stacks on a board?

Yes that is also possible. :wink: if it doesn't work for you please open a separate issue for that.

**Comment by GoetheG** [comment]
This is a really necessary feature. With that you can really start competing with Trello :). My hope is that open source software will be key in the future. Adding this feature would bring us one step forward.

Thank you guys!

**Comment by juliushaertl** [comment]
Some details that need to be clarified before implementing:

- Labels that are assigned to cards are specific to the individual boards. We need to handle them somehow, since the other board doesn't have those labels. I'd just drop them to keep it simple.
- Same for assigned users. Here we could just keep the users that also have access to the other board.

**Comment by ashgoodman** [comment]
+1, this was the first thing I noticed I needed that this excellent software doesn't already provide!

**Comment by Nebri** [comment]
@juliushaertl Where would you like this to be placed? I'm thinking the side panel on the card details with typeahead's for the new board?

**Comment by juliushaertl** [comment]
@Nebri It would make sense to add this as an option to the 3-dots menu on the board view. We can then just show a popover with the option to copy/move the card to a different stack or board. Similar to how trello does it:

![image](https://user-images.githubusercontent.com/3404133/44512992-7a41df80-a6bc-11e8-9d66-21290170fc10.png)

**Comment by Nebri** [comment]
Sorry for inactivity, needed a break / interviews. Getting back on this task.

**Comment by Nebri** [comment]
@juliushaertl Having a hard time figuring out how to do the pop over form you have mocked up there. Anybody able to contribute a code snippet to help get me on my way?

**Comment by Nebri** [comment]
Aplogies for inactivity, resuming work on this task now :).

**Comment by Nebri** [comment]
@nextcloud/deck opened up PR #652 as a WIP. I'm blocked on trying to resolve the entire boardsList. If anyone doesn't mind looking at that PR and the changes I've got in the boardscontroller that will help to get this done in a timely manner.

My guess is that I am either misuing the BoardService or it is bugged and not returning the expected results. I need to which is the case in order to continue.

**Comment by brianbgness** [comment]
I would love to have this functionality!  If too difficult to implement drag and drop, perhaps a Move option could be added to the card triple-dot menu.  Then be prompted for which board and stack to move to.

**Comment by RyanFletcher86** [comment]
Yes the ... dot menu is my plan for this. I've been taking some job interviews in my local area and getting bogged down by the code challenges companies are wanting me to do as part of their application process. Going to touching up my own portfolio tomorrow morning, but back around to this in the afternoon.

**Comment by Aesculapius** [comment]
Anyone else working on this? It sounds like an easy added feature which solves a lot of hassle when moving cards over boards...

**Comment by RyanFletcher86** [comment]
> Anyone else working on this? It sounds like an easy added feature which solves a lot of hassle when moving cards over boards...

JuliusHeartl is currently assigned to this which is an indication that he is actively working on it. There's been a lot of overhead work that contributors have been working towards that needs to be done first (the migration to vue.js being the largest).

I've also been bogged down by a real world job search, hence why I released the issue from my personal custody.

**Comment by juliushaertl** [comment]
Yes, we definitely should get the vue migration done before we continue with feature development, otherwise we end up doing the work twice. Unfortunately I haven't had a lot of time recently for deck, I hope that will change in the next weeks. :wink:

**Comment by rhersel** [comment]
I support this request as well. This is a basic feature to work seriously with Kanban boards.

**Comment by oskiebe** [comment]
+1 - would love to see this!

**Comment by juliushaertl** [comment]
@oskiebe @rhersel Please use the [Github Reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) feature instead of commenting that also need a specific feature. That way we can see your support without notifying all users subscribed to a given issue.

**Comment by juliushaertl** [comment]
Closing as this was implemented in the vue branch and is now merged to master. 

@jakobroehrl There is a bounty on it, make sure to pick it up :wink:

**Comment by treborxy** [comment]
Hello juliushaertl,
I'm not sure: is "moving cards from one board to another" implemented in branch master? And if, how to use i?
Thx, Robert

**Comment by jakobroehrl** [comment]
@treborxy 
It's implented. Use it with the 3 dot menu of any card

**Comment by juliushaertl** [comment]
Note this is not yet released in a stable version, just available in the nightly builds.

**Comment by treborxy** [comment]
Thanks! I tried to test nighly and uploaded the decomp. tar - but what I get in protocol after activating is an error - could you please give a hint how to prepare the nightlies? Thx a lot!

`Exception: Cannot include autoload. Did you run install dependencies using composer?

    /www/htdocs/[...]/lib/private/legacy/app.php - line 266:
    require_once()
    /www/htdocs/[...]/lib/private/legacy/app.php - line 155:
    OC_App::requireAppFile("deck")
    /www/htdocs/[...]/lib/private/legacy/app.php - line 128:
    OC_App::loadApp("deck")
    /www/htdocs/[...]/apps/dav/appinfo/app.php - line 35:
    OC_App::loadApps([ "dav"])
    /www/htdocs/[...]/lib/private/legacy/app.php - line 266:
    require_once("/www/htdocs ... p")
    /www/htdocs/[...]/lib/private/legacy/app.php - line 155:
    OC_App::requireAppFile("dav")
    /www/htdocs/[...]/lib/private/legacy/app.php - line 128:
    OC_App::loadApp("dav")
    /www/[...]/lib/base.php - line 993:
    OC_App::loadApps()
    /www/htdocs/[...]/index.php - line 38:
    OC::handleRequest()`

**Comment by juliushaertl** [comment]
Oh, sorry for the inconvenience here. It seems that the nightly build upload is currently failing. I'll look into that and created a separate issue for tracking https://github.com/nextcloud/deck/issues/1595

**Comment by fabianski7** [comment]
Hello,

Sorry for reliving this topic but is there any way to move more than one card simultaneously? Something like joining two boards into one?

Thanks.
